source/includes/v2/_captions.md

```diff
@@ -0,0 +1,319 @@
+  'Content-Type': "application/json",
+  'Authorization': "Basic {token}",
+}
+
+response = requests.request("POST", url, data=payload, headers=headers)
+
+print(response.text)
+```
+
+```javascript
+// NodeJS
+
+var request = require("request");
+
+var options = {
+  method: 'POST',
+  url: 'https://api.vidgrid.com/v2/captions',
+  headers: { 
+    Authorization: 'Basic {token}',
+    'Content-Type': 'application/json' 
+  },
+  body: { 
+    video_identifier: "...",
+    type: "machine"
+  },
+  json: true 
+};
+
+request(options, function (error, response, body) {
+  if (error) throw new Error(error);
+
+  console.log(body);
+});
+```
+
+> Example create caption response. See [Caption Resource](#caption-resource) for more details.
+
+```json
+{
+  "data": Caption Resource Object
+}
+```
+
+`POST https://api.vidgrid.com/v2/captions`
+
+### HTTP Parameters
+
+| Param | Type | Description | Default |
+| ----- | ---- | ----------- | ------- |
+| **video_identifier** | string | The unique identifier of the video you are requesting a caption for. | *Required* |
+| **type** | string | The type of caption you would like to request.<br>*Possible values: `machine`, `professional`* | *Required* |
+
+## Retrieve Caption
+
+This endpoint returns an array of [Caption Resources](#caption-resource).
+
+### HTTP Request
+
+> Example retrieve caption request.
+
+```shell
+curl -X GET \
+  'https://api.vidgrid.com/v2/captions' \
+  -H 'Authorization: Basic {token}' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "identifiers": [
+      "...",
+      "..."
+    ]
+  }'
+```
+
+```ruby
+require 'uri'
+require 'net/http'
+
+url = URI("https://api.vidgrid.com/v2/captions")
+
+http = Net::HTTP.new(url.host, url.port)
+
+request = Net::HTTP::Get.new(url)
+request["Content-Type"] = 'application/json'
+request["Authorization"] = 'Basic {token}'
+request.body = '{
+  "identifiers": [
+    "...",
+    "..."
+  ]
+}'
+
+response = http.request(request)
+puts response.read_body
+```
+
+```python
+import requests
+
+url = "https://api.vidgrid.com/v2/captions"
+
+payload = '{
+  "identifiers": [
+    "...",
+    "..."
+  ]
+}'
+headers = {
+  'Content-Type': "application/json",
+  'Authorization': "Basic {token}",
+}
+
+response = requests.request("GET", url, data=payload, headers=headers)
+
+print(response.text)
+```
+
+```javascript
+// NodeJS
+
+var request = require("request");
+
+var options = {
+  method: 'GET',
+  url: 'https://api.vidgrid.com/v2/captions',
+  headers: { 
+    Authorization: 'Basic {token}',
+    'Content-Type': 'application/json' 
+  },
+  body: { 
+    identifiers: [
+      '...',
+      '...'
+    ]
+  },
+  json: true 
+};
+
+request(options, function (error, response, body) {
+  if (error) throw new Error(error);
+
+  console.log(body);
+});
+```
+
+> Example retrieve caption response. See [Caption Resource](#caption-resource) for more details.
+
+```json
+{
+  "data": [
+    Caption Resource Object,
+    Caption Resource Object
+  ]
```

Comment:
is this still accurate with the changes that we made with dlo that one day? we got rid of the `toArray` helper method on the mapper abstract class. I thought for singular requests it was just going to serialize the object and return that... so it wouldn't be in an array.

I also seem to remember that we have code that will always put it in an array. I don't quite remember

Reply:
I believe it's still accurate but will verify

Reply:
Still accurate